### ISSUE TITLE
Style/AlignHash conflicts with Style/AlignParameters

### ISSUE THREAD

**Issue by adamgotterer** [rubocop-hq/rubocop#3803]
## Expected behavior

Given the following .rubocop.yml and method call I expected there not to be a Style/AlignHash error.

```
# .rubocop.yml
Style/AlignParameters:
  EnforcedStyle: with_first_parameter

Style/AlignHash:
  EnforcedHashRocketStyle: key
```

```ruby
somemethod(a: 'val'
  b: 'val1')
```

## Actual behavior

`Style/AlignHash: Align the elements of a hash literal if they span more than one line`. It seems that if the parameters are a hash that it uses AlignHash instead of AlignParameters. As best I can tell there isn't any alignment settings that can be set in AlignHash?

## RuboCop version

0.46.0 (using Parser 2.3.3.1, running on ruby 2.3.1 x86_64-linux)

**Comment by bbatsov** [comment]
The problem is that you can't know if this is a final hash or keyword params for sure. Probably in such cases we should treat this as keyword params to avoid getting into such odd situations. @jonas054 thoughts?

**Comment by jonas054** [comment]
So the problem is that the configuration
```yaml
Style/AlignParameters:
  EnforcedStyle: with_fixed_indentation
```
gives the inconsistency
```ruby
somemethod(a: 'val'
           b: 'val1')
somemethod('val'
  'val1')
```
We could let `Style/AlignHash` leave the inspection of braceless hash arguments to `Style/AlignParameters`, meaning that we view the elements as individual keyword arguments rather than key/value pairs in a hash. That would give us
```ruby
somemethod(a: 'val'
  b: 'val1')
somemethod('val'
  'val1')
```
which I guess is more consistent and logical. The only problem I foresee is that the configuration
```yaml
Style/AlignParameters:
  EnforcedStyle: with_first_parameter

Style/AlignHash:
  EnforcedColonStyle: table
```
which today gives
```ruby
somemethod(a:    'val'
           bbbb: 'val1')
hash = {
  a:    'val'
  bbbb: 'val1')
}
```
would give
```ruby
somemethod(a: 'val'
           bbbb: 'val1')
hash = {
  a:    'val'
  bbbb: 'val1')
}
```
i.e., you wouldn't be able to have `table` (or `separator`) alignment in method calls.

So it's a tradeoff. I'm leaning towards making the change, but don't have a strong opinion.

Or is there another solution where we cold have the cake and eat it too?

**Comment by rdallasgray** [comment]
I'm having a similar issue with this. With default settings:
```ruby
# rubocop: 'Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.'
validates :limit, numericality: {
            less_than_or_equal_to: MAX_LIMIT,
            allow_blank: true
          }

# rubocop: ''
validates(:limit, numericality: {
            less_than_or_equal_to: MAX_LIMIT,
            allow_blank: true
          })
```

**Comment by jonas054** [comment]
@rdallasgray It's not quite the same problem, since the stuff between the curly braces can only be regarded as a hash, and not as two method parameters. You get a report from `Layout/IndentHash`, because it wants the following indentation, depending on configuration:

```ruby
# EnforcedStyle: special_inside_parentheses (default)
validates :limit, numericality: {
  less_than_or_equal_to: MAX_LIMIT,
  allow_blank: true
}

validates(:limit, numericality: {
            less_than_or_equal_to: MAX_LIMIT,
            allow_blank: true
          })

# EnforcedStyle: consistent
validates :limit, numericality: {
  less_than_or_equal_to: MAX_LIMIT,
  allow_blank: true
}

validates(:limit, numericality: {
  less_than_or_equal_to: MAX_LIMIT,
  allow_blank: true
})

# EnforcedStyle: align_braces
validates :limit, numericality: {
                                  less_than_or_equal_to: MAX_LIMIT,
                                  allow_blank: true
                                }

validates(:limit, numericality: {
                                  less_than_or_equal_to: MAX_LIMIT,
                                  allow_blank: true
                                })
```
The style you prefer is not among those supported, I guess.
